Sources/Amplitude/Constants.swift

```diff
@@ -49,8 +48,27 @@ public struct Constants {
     static let AMP_TRACKING_OPTION_PLATFORM = "platform"
     static let AMP_TRACKING_OPTION_REGION = "region"
     static let AMP_TRACKING_OPTION_VERSION_NAME = "version_name"
+
+    static let AMP_AMPLITUDE_PREFIX = "[Amplitude] "
+
     static let AMP_SESSION_END_EVENT = "session_end"
     static let AMP_SESSION_START_EVENT = "session_start"
+    static let AMP_APPLICATION_INSTALLED_EVENT = "\(AMP_AMPLITUDE_PREFIX)Application Installed"
+    static let AMP_APPLICATION_UPDATED_EVENT = "\(AMP_AMPLITUDE_PREFIX)Application Updated"
+    static let AMP_APPLICATION_OPENED_EVENT = "\(AMP_AMPLITUDE_PREFIX)Application Opened"
+    static let AMP_APPLICATION_BACKGROUNDED_EVENT = "\(AMP_AMPLITUDE_PREFIX)Application Backgrounded"
+    static let AMP_DEEP_LINK_OPENED_EVENT = "\(AMP_AMPLITUDE_PREFIX)Deep Link Opened"
+    static let AMP_SCREEN_VIEWED_EVENT = "\(AMP_AMPLITUDE_PREFIX)Screen Viewed"
```

Comment:
I am seeing `Screen Viewed` more than expected in the sample app. Opening the app with no other actions gives this output:

```
2023-09-13 11:04:22.425642-0700 AmplitudeSwiftUIExample[98737:14915236] [Logging] Debug: Processed event: Optional("{\"dma\":null,\"session_id\":1694627760840,\"region\":null,\"device_id\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"version_name\":\"1.0\",\"os_version\":\"16.4\",\"platform\":\"iOS\",\"country\":null,\"device_model\":\"Simulator\",\"carrier\":null,\"event_properties\":{\"[Amplitude] Version\":\"1.0\",\"[Amplitude] From Background\":true,\"[Amplitude] Build\":\"1\"},\"insert_id\":\"FFD50847-0437-40E1-92FD-B9CB3F031940\",\"revenue\":null,\"event_id\":108,\"partner_id\":null,\"location_lng\":null,\"os_name\":\"ios\",\"event_type\":\"[Amplitude] Application Opened\",\"revenue_type\":null,\"quantity\":null,\"city\":null,\"adid\":null,\"price\":null,\"product_id\":null,\"idfv\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"device_brand\":null,\"idfa\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"user_id\":\"justin.fiedler@amplitude.com\",\"location_lat\":null,\"device_manufacturer\":\"Apple\",\"ip\":\"$remote\",\"time\":1694628262421,\"language\":\"en\",\"app_version\":null,\"library\":\"amplitude-swift\\/0.5.0\"}")
2023-09-13 11:04:22.465946-0700 AmplitudeSwiftUIExample[98737:14915236] [Logging] Debug: Processed event: Optional("{\"dma\":null,\"session_id\":1694627760840,\"region\":null,\"device_id\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"version_name\":\"1.0\",\"os_version\":\"16.4\",\"platform\":\"iOS\",\"country\":null,\"device_model\":\"Simulator\",\"carrier\":null,\"event_properties\":{\"[Amplitude] Screen Name\":\"<UICompatibilityInput: 0x7fe903886400>\"},\"insert_id\":\"1BBE3855-B559-4D9D-BAFE-F1524DB15DBC\",\"revenue\":null,\"event_id\":109,\"partner_id\":null,\"location_lng\":null,\"os_name\":\"ios\",\"event_type\":\"[Amplitude] Screen Viewed\",\"revenue_type\":null,\"quantity\":null,\"city\":null,\"adid\":null,\"price\":null,\"product_id\":null,\"idfv\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"device_brand\":null,\"idfa\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"user_id\":\"justin.fiedler@amplitude.com\",\"location_lat\":null,\"device_manufacturer\":\"Apple\",\"ip\":\"$remote\",\"time\":1694628262462,\"language\":\"en\",\"app_version\":null,\"library\":\"amplitude-swift\\/0.5.0\"}")
2023-09-13 11:04:22.477126-0700 AmplitudeSwiftUIExample[98737:14915236] [Logging] Debug: Processed event: Optional("{\"dma\":null,\"session_id\":1694627760840,\"region\":null,\"device_id\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"version_name\":\"1.0\",\"os_version\":\"16.4\",\"platform\":\"iOS\",\"country\":null,\"device_model\":\"Simulator\",\"carrier\":null,\"event_properties\":{\"[Amplitude] Screen Name\":\"<UIEditingOverlay: 0x7fe902222310>\"},\"insert_id\":\"7EF62E41-E36D-4212-B3E6-8F7E3A2930EF\",\"revenue\":null,\"event_id\":110,\"partner_id\":null,\"location_lng\":null,\"os_name\":\"ios\",\"event_type\":\"[Amplitude] Screen Viewed\",\"revenue_type\":null,\"quantity\":null,\"city\":null,\"adid\":null,\"price\":null,\"product_id\":null,\"idfv\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"device_brand\":null,\"idfa\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"user_id\":\"justin.fiedler@amplitude.com\",\"location_lat\":null,\"device_manufacturer\":\"Apple\",\"ip\":\"$remote\",\"time\":1694628262473,\"language\":\"en\",\"app_version\":null,\"library\":\"amplitude-swift\\/0.5.0\"}")
2023-09-13 11:04:22.544407-0700 AmplitudeSwiftUIExample[98737:14915236] [Logging] Debug: Processed event: Optional("{\"dma\":null,\"session_id\":1694627760840,\"region\":null,\"device_id\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"version_name\":\"1.0\",\"os_version\":\"16.4\",\"platform\":\"iOS\",\"country\":null,\"device_model\":\"Simulator\",\"carrier\":null,\"event_properties\":{\"[Amplitude] Screen Name\":\"<UICompatibilityInput: 0x7fe903886400>\"},\"insert_id\":\"78EC0C68-E2F9-4522-BD09-909CA3C91BA7\",\"revenue\":null,\"event_id\":111,\"partner_id\":null,\"location_lng\":null,\"os_name\":\"ios\",\"event_type\":\"[Amplitude] Screen Viewed\",\"revenue_type\":null,\"quantity\":null,\"city\":null,\"adid\":null,\"price\":null,\"product_id\":null,\"idfv\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"device_brand\":null,\"idfa\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"user_id\":\"justin.fiedler@amplitude.com\",\"location_lat\":null,\"device_manufacturer\":\"Apple\",\"ip\":\"$remote\",\"time\":1694628262540,\"language\":\"en\",\"app_version\":null,\"library\":\"amplitude-swift\\/0.5.0\"}")
2023-09-13 11:04:22.556197-0700 AmplitudeSwiftUIExample[98737:14915236] [Logging] Debug: Processed event: Optional("{\"dma\":null,\"session_id\":1694627760840,\"region\":null,\"device_id\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"version_name\":\"1.0\",\"os_version\":\"16.4\",\"platform\":\"iOS\",\"country\":null,\"device_model\":\"Simulator\",\"carrier\":null,\"event_properties\":{\"[Amplitude] Screen Name\":\"<UIEditingOverlay: 0x7fe902222310>\"},\"insert_id\":\"F6DFC2DE-10EC-40D5-BEF3-B7554FC37EEC\",\"revenue\":null,\"event_id\":112,\"partner_id\":null,\"location_lng\":null,\"os_name\":\"ios\",\"event_type\":\"[Amplitude] Screen Viewed\",\"revenue_type\":null,\"quantity\":null,\"city\":null,\"adid\":null,\"price\":null,\"product_id\":null,\"idfv\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"device_brand\":null,\"idfa\":\"46D758FB-CFEE-41DD-A0CB-4A3A6E9D1C31\",\"user_id\":\"justin.fiedler@amplitude.com\",\"location_lat\":null,\"device_manufacturer\":\"Apple\",\"ip\":\"$remote\",\"time\":1694628262552,\"language\":\"en\",\"app_version\":null,\"library\":\"amplitude-swift\\/0.5.0\"}")
```

Reply:
After latest changes I am not seeing to few Screen Views. I was expecting a screen view per app opened.

<img width="639" alt="image" src="https://github.com/amplitude/Amplitude-Swift/assets/5790872/ec7a58d8-6106-4e84-bb0d-644df92ca8ff">

<img width="667" alt="image" src="https://github.com/amplitude/Amplitude-Swift/assets/5790872/6988db2b-a284-40cb-a6ac-4db1482c4fc8">

Reply:
1. Current Amplitude-Swift implementation is based on Amplitude-iOS code.
Differences:
Amplitude-iOS: default Amplitude instance is always used:
```
[[Amplitude instance] logEvent:kAMPScreenViewed ...
```
Amplitude-Swift: track screen events for all Amplitude instances with `defaultTracking.screen == true`: 
```
for amplitude in UIKitScreenViews.amplitudes {
    amplitude.track(eventType: Constants.AMP_SCREEN_VIEWED_EVENT, eventProperties: eventProperties)
}
```
As mentioned in [our docs](https://amplitude.atlassian.net/wiki/spaces/GROWTH/pages/2109079759/Test+Mobile+SDK+Default+Events#iOS-Objective-C-SDK) only UIKit is supported:
> Note that as SwiftUI creates layers of controller wrappers, it might fire multiple screen view events in the same screen with the trigger of viewDidAppear. There seems no good support of the auto screen view tracking at the moment.

Currently `defaultTracking.screen == true` should be used only for UIKit applications. In SwiftUI applications screen views should be tracked manually (maybe we should add `amplitude.screen(...)` method). 

2. It looks like current Amplitude-iOS/Amplitude-Swift implementation is inspired by

Reply:
Thanks @falconandy 
1. Couple of things
  a. Screen view limitations in SwiftUI - I made a [ticket to update the public docs](https://amplitude.atlassian.net/browse/AMP-84634)
  b. `screen()` method. I want to keep the top-level interface as small as possible. I would prefer a `ScreenViewedEvent` that can be sent with `track()` instead.
2. I think that makes sense to add. Do you agree @liuyang1520?
3. TIL. Thank you

Reply:
- +1 on 1.b, to reuse the track method if possible.
- 2, make sense, pls do!
- Good to learn it. I feel it is ok to not fire the screen viewed event when opening from background. Similar to browser Page Viewed event, we are not firing new page view event when switching tabs or bring the browser to foreground, as it is not a new page. What do you think? @justin-fiedler